src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -252,6 +257,41 @@ private class Emit(
       case ArrayLen(a) =>
         val (doa, ma, va) = emit(a)
         (doa, ma, TContainer.loadLength(region, coerce[Long](va)))
+      case x@ArrayRange(startir, stopir, stepir) =>
+        val rangem = irmethods.getOrElseUpdate("ArrayRange", {
```

Comment:
I'd remove irmethods for now.

Reply:
so just inline ArrayRange for now?

Reply:
Yes.  It actually needs to get deforested, so sticking it in a function is going in the wrong direction.  Range => for (i = start; i < end; i += step) ... and Filter => if (...) ...